### DISCUSSION TITLE
We need your feedback! OpenTelemetry in Powertools for AWS

### DISCUSSION THREAD

**Discussion by willfarrell** [aws-powertools/powertools-lambda-typescript#665] [General]
> [!Note]
> Update from Powertools for AWS Team: we're actively considering adding OpenTelemetry support, please head over to aws-powertools/powertools-lambda#90 and let us know what you think!

---

Recently ran into AWS Distro for OpenTelemetry (https://aws-otel.github.io/docs/getting-started/javascript-sdk) that also does Traces, Metrics, and Logs for AWS Lambda. Curious how this relates to powertools typescript, if at all.

https://opentelemetry.io/status/

**Comment by flochaz** [comment]
Hi thanks for opening this discussion, following what @heitorlessa in our [public slack channel](https://awsdevelopers.slack.com/archives/C01A6KK4UFK/p1647333343851399?thread_ts=1647012398.063549&cid=C01A6KK4UFK), we do consider evaluating OTel in the future, just not in the foreseeable future (e.g., GA, operational excellence, feature parity, [OTel specification stability](https://opentelemetry.io/status/)).

**Comment by willfarrell** [comment]
Does this mean it's GA? https://aws.amazon.com/about-aws/whats-new/2022/05/aws-distro-opentelemetry-metric-support/

**Reply by dreamorosi** [reply]
Each module (i.e. metrics, logger, etc.) follows its own lifecycle and development ([see status page](https://opentelemetry.io/status/)).

The announcement you've linked refers to metrics specifically reaching GA.

According to the page I linked the one still in draft is Logging at the moment.

**Comment by willfarrell** [comment]
I'm working on Middy@4.0.0 (https://github.com/middyjs/middy/pull/908) and updating how X-Ray is integrated. While reviewing the X-Ray SDK docs I noticed they plan to deprecate it:

> AWS X-Ray recommends using AWS Distro for OpenTelemetry (ADOT) to instrument your application instead of this X-Ray SDK due to its wider range of features and instrumentations.

~ https://github.com/aws/aws-xray-sdk-node

I was a little surprised they recommended ADOT over Powertools. For tracing, powertools is more inline with the X-Ray SDK, making it easier on my end ... but now I'm not following AWS own recommendations. Not confusing at all.

Has using ADOT been discussed recently? Is the long term goal to standardize on ADOT, and have powertools simplify using it (i.e. https://aws-otel.github.io/docs/getting-started/js-sdk/trace-manual-instr#sending-traces-to-aws-x-ray is not dev friendly). Maybe plans of a staged rollout? Should the X-Ray SDK notice be updated to include powertools?

Sorry for all the questions, I just want to make sure the change I make won't need to change next year.

**Comment by heitorlessa** [comment]
Hi Will, that's no longer the case (X-Ray SDK deprecation) - we discussed
it internally last week. As noted by customers, there are performance
challenges in OTel that are currently being reviewed and no ETA atm. This
means, we won't be making any plans for moving to OTel at least in 2023.
We'll communicate with ample time if anything changes.

**Comment by dreamorosi** [comment]
We're actively considering adding OpenTelemetry support, please head over to aws-powertools/powertools-lambda#90 and let us know what you think!
